app/services/articles/feed.rb

```diff
@@ -137,10 +139,10 @@ def globally_hot_articles(user_signed_in)
         order("hotness_score DESC")
       featured_story = hot_stories.where.not(main_image: nil).first
       if user_signed_in
-        offset = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 2, 2, 2, 3, 3, 4, 5, 6, 7, 8, 9, 10, 11].sample # random offset, weighted more towards zero
+        offset = RANDOM_OFFSET_VALUES.reject { |i| i >= hot_stories.count }.sample # random offset, weighted more towards zero
```

Comment:
The amount of hoop jumping I'm doing for the `offset` is getting crazy. This field has been a constant source of issues: do we even want to keep it?

Reply:
I'd find it slightly more readable as 

```ruby
a.select { |i| i < hot_stories.count }.sample
```

but I agree that the whole concept is a bit confusing.

Reply:
BTW, another way you could write this is 

```ruby
RANDOM_OFFSET_VALUES[0..RANDOM_OFFSET_VALUES.index(hot_stories.count)].sample
```

since an upper bound of `nil` will return the full array. 

```ruby
RANDOM_OFFSET_VALUES[0..nil]
#=> [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 2, 2, 2, 3, 3, 4, 5, 6, 7, 8, 9, 10, 11]
```

If the `hot_stories.count` is > 11 `index` will traverse the whole array once, but so will `reject`/`select`. Otherwise this does slightly less work while still being readable (IMHO).

Reply:
> I'd find it slightly more readable as
> 
> ```ruby
> a.select { |i| i < hot_stories.count }.sample
> ```

Oh agreed, I quite like that!

Reply:
> BTW, another way you could write this is
> 
> ```ruby
> RANDOM_OFFSET_VALUES[0..RANDOM_OFFSET_VALUES.index(hot_stories.count)].sample
> ```
> 
> since an upper bound of `nil` will return the full array.
> 
> ```ruby
> RANDOM_OFFSET_VALUES[0..nil]
> #=> [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 2, 2, 2, 3, 3, 4, 5, 6, 7, 8, 9, 10, 11]
> ```
> 
> If the `hot_stories.count` is > 11 `index` will traverse the whole array once, but so will `reject`/`select`. Otherwise this does slightly less work while still being readable (IMHO).

I'm less wild about this solution as it depends on the offset array composition staying the same and never dropping below a certain point and not skipping values. I don't think we'd ever have an offset array like `[0, 0, 3, 3, 6, 6]`, but it's undocumented and feels bad to me!